hd-wallet/pluginv2.lua

```diff
@@ -0,0 +1,511 @@
+-- - Import master key in SDKMS as secret raw key
+-- 
+-- ## Input/Output JSON object format
+-- **Input**
+-- ```
+-- {
+--     "master_key_id": "<Master-Key-UUID>",
+--     "path": "<Child-Key-Path>"  ,
+--     "msg_hash": "<32-Byte-Message-Hash>",
+--     "coin": "<Coin-Type>"
+-- }
+-- ```
+-- **Output**
+-- ```
+--  "xprv": "<HD-Wallet-Private-Key>",
+--  "xpub": "<HD-Wallet-Public-Key>",
+--  "coin_signature": "<Bitcoin-canonicalized-ECDSA-signature>",
+--  "signature": "<ECDSA signature>"
+-- ```
+-- * `master_key_id`: UUID of master key imported in SDKMS
+-- * `path`: Path of key to be derived to sign e.g: m/0, m/1, m/2/10 etc
+-- * `msg_hash`: 32 byte SHA-3 message hash
+-- * `coin`: coin type utxo or eth
+-- * `xprv`: BIP0032 private key
+-- * `xpub`: BIP0032 public key
+-- * `coin_signature`: Bitcoin canonicalized ECDSA signature
+-- * `signature`: ECDSA signature
+-- ## Example Input/Output JSON object for UTXO COIN (BTC, LTC, BCH, etc.)
+-- **Input JSON object**
+-- ```
+-- {
+--    "master_key_id": "0eae8ff0-553e-4f47-bb64-7c87f34bf5e5",
+--    "coin": "utxo",
+--    "path": "m/2",
+--    "msg_hash": "45a0ee821b05400f513891bbb567a99139f3df72e9e1d4b48186841cc5996d2f"
+-- }
+-- ```
+-- **Output JSON object**
+-- ```
+-- {
+--   "xprv": "xprv9uZghWCSYwDho7us3q1WLBjVYx2xzVJNT8qNo4P9i8wa3tQJYbffzztTF6wXjuorG49NXahqraWsrVUmy3uTJLkvSYXyDLnHHU1GJibUk2t",
+--   "xpub": "xpub68Z371jLPJn11bzL9rYWhKgE6ysTPx2DpMkybSnmGUUYvgjT68yvYoCw6PP8Vo7YoZRC6iqrfpixEUG694KgHPYYnydGuEYDwjESStYxYxe",
+--   "signature": "3045022100af9bf94c4959328b56861ca5f175b5e59014cb5bd2a5fcee2e95b1563dbc652e0220411ff01751af64d6b7209908fc58f527b07a0a9258eee7be7aa5704136954b02",
+--   "coin_signature": "af9bf94c4959328b56861ca5f175b5e59014cb5bd2a5fcee2e95b1563dbc652e411ff01751af64d6b7209908fc58f527b07a0a9258eee7be7aa5704136954b02"
+-- }
+-- ```
+-- ## Example Input JSON object for ETH
+-- ```
+-- {
+--    "masterKeyId": "5aef3e3f-7927-49b2-b252-bf84b6980f95",
+--    "coin": "eth",
+--    "msgHash": "45a0ee821b05400f513891bbb567a99139f3df72e9e1d4b48186841cc5996d2f"
+-- }
+-- ```
+-- ## Example Input JSON object for XPRV 
+-- ```
+-- {
+--    "import": true,
+--    "xprvEncId": "5aef3e3f-7927-49b2-b252-bf84b6980f95",
+--    "wrapKeyId": "02e5c697-87cd-45b6-8163-957fa5e13370",
+--    "name": "Imported Key"
+-- }
+-- ```
+-- ## References
+-- - https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
+-- - https://xrpl.org/sign.html
+-- ### Release Notes
+--  V 1.0 
+--  - Initial Release
+--  V 2.0 
+--  - Added XRP Transaction Signature Support
+--  - Added Private Key flag and checking
+--  - Added Eth, BCH, BTC test cases
+---------- @@@@@@@@@@@@@@@@@@@@@@@@@ ----------
+----------      COMMON UTILITIES     ----------
+---------- @@@@@@@@@@@@@@@@@@@@@@@@@ ----------
+----------------- Debug Logging ---------------
+local DEBUG = {}
+local DEBUG_N = 1
+function debug(msg)
+  DEBUG[DEBUG_N] = msg
+  DEBUG_N = DEBUG_N + 1
+end
```

Comment:
I think there was a misunderstanding about linebreaks. 
There should be line breaks between sections and functions! This is currently not readable:
```
-- return public key from private key 
function public_key_for_private_key(key_byte)
    local point = get_point_coordinates_from_private_key(key_byte)
    return compress_public_key(string.sub(point, 1, 64), string.sub(point, 65, 128))
end
-- parse input path
function parse_path(child_path)
    local path_table = split(child_path, "/")
    return path_table
end
```
So please add linebreaks where they belong.

Reply:
And also, please remove all introducing trailing whitespace (e.g., see `-- return public key from private key ` just above)